wavefront/server/apps/floconsole/floconsole/services/floware_proxy_service.py

```
@@ -56,15 +60,28 @@ async def proxy_request(
+        session = request.state.session
+        user_id = UUID(session.user_id)
 
-        # Step 2: Fetch app details from database
+        # Step 2: Check if user has access to app (RBAC)
+        try:
+            has_access = await self.user_service.check_user_has_app_access(
+                user_id=user_id, app_id=UUID(app_id)
+            )
+
+            if not has_access:
+                raise ValueError('User does not have access to this app')
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)